Geometry/HGCalMapping/plugins/alpaka/HGCalMappingCellESProducer.cc

```diff
@@ -41,9 +45,49 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE {
         desc.add<std::vector<std::string> >("filelist", std::vector<std::string>({}))
             ->setComment("list of files with the readout cells of each module");
         desc.add<edm::ESInputTag>("cellindexer", edm::ESInputTag(""))->setComment("Dense cell index tool");
+        desc.add<edm::FileInPath>(
+                "offsetfile",
+                edm::FileInPath("Geometry/HGCalMapping/data/CellMaps/calibration_to_surrounding_offsetMap.txt"))
+            ->setComment("file containing the offsets between calibration and surrounding cells");
         descriptions.addWithDefaultLabel(desc);
       }
 
+      std::map<std::tuple<std::string, int, int>, int> makeOffsetMap(edm::FileInPath input_offsetfile) {
+        std::map<std::tuple<std::string, int, int>, int> offsetMap;
+        const auto& offsetfile = input_offsetfile.fullPath();
+        std::ifstream stream_offsetfile(offsetfile);
+        std::string line;
+        // Skip the first line (description of column content)
+        std::getline(stream_offsetfile, line);
+        while (std::getline(stream_offsetfile, line)) {
+          std::istringstream iss(line);
+          std::string r_typecode_str, r_chip_str, r_half_str, offset_str;
+          if (!(iss >> r_typecode_str >> r_chip_str >> r_half_str >> offset_str)) {
+            edm::LogError("HGCalMappingCellESProducer") << "Error reading offset file" << std::endl;
+            break;
+          }
+
+          int r_chip, r_half, offset;
+          auto parse_entry = [](const std::string& str, int& value) -> bool {
+            const char* begin = str.data();
+            const char* end = begin + str.size();
+            auto result = std::from_chars(begin, end, value);
+            return result.ec == std::errc();
+          };
+
+          if (!parse_entry(r_chip_str, r_chip) || !parse_entry(r_half_str, r_half) ||
+              !parse_entry(offset_str, offset)) {
+            edm::LogError("HGCalMappingCellESProducer")
+                << "Error parsing offset file entries (typecode, roc, halfroc, and offset): " << r_typecode_str << ", "
+                << r_chip_str << ", " << r_half_str << ", " << offset_str << "\n";
+            break;
+          }
+
+          offsetMap[std::make_tuple(r_typecode_str, r_chip, r_half)] = offset;
```

Comment:
here instead of `r_typecode_str` you could re-use, as @fwyzard proposes, `auto typeidx = cellIndexer.getEnumFromTypecode(r_typecode_str);`